FF13Rando/Randomizers/CrystariumRando.cs

```diff
@@ -412,23 +412,42 @@ private string GetFirstRole(string c)
 
             chars.ForEach(name =>
             {
+                Dictionary<Role, (int, int, int)> roleTotals = crystariums[name].Values.Where(c => !c.ID.StartsWith("!")).GroupBy(c => c.iRole).Where(g => g.Key != Role.None).OrderBy(g => g.Key).Select(roleGroup =>
+                {
+                    var hpTotal = roleGroup.Where(c => c.iType == CrystariumType.HP).Sum(c => c.iValue);
+                    var strengthTotal = roleGroup.Where(c => c.iType == CrystariumType.Strength).Sum(c => c.iValue);
+                    var magicTotal = roleGroup.Where(c => c.iType == CrystariumType.Magic).Sum(c => c.iValue);
+                    return (roleGroup.Key, (hpTotal, strengthTotal, magicTotal));
+                }).ToDictionary(kv => kv.Item1, kv => kv.Item2);
+                int charHpTotal = roleTotals.Values.Select(t => t.Item1).Sum();
+                int charStrTotal = roleTotals.Values.Select(t => t.Item2).Sum();
+                int charMagTotal = roleTotals.Values.Select(t => t.Item3).Sum();
+                Func<(int, int, int), string> writeStatsForDisplay = ((int, int, int) input) =>
+                    string.Join("<br/>", $"HP + {input.Item1}", $"Strength + {input.Item2}", $"Magic + {input.Item3}");
                 page.HTMLElements.Add(new Table(name[0].ToString().ToUpper() + name.Substring(1),
-                    new string[] { "Stage", "Commando", "Ravager", "Sentinel", "Synergist", "Saboteur", "Medic" }.ToList(),
-                    new int[] { 10, 15, 15, 15, 15, 15, 15 }.ToList(),
+                    new string[] { "Stage", "Commando", "Ravager", "Sentinel", "Synergist", "Saboteur", "Medic", "Totals" }.ToList(),
+                    new int[] { 4, 14, 14, 14, 14, 14, 14, 12 }.ToList(),
```

Comment:
I can't remember if the docs required the weights to add up to 100...
if they do require that, I'll change that just it's not as weird to work with.